src/tools/convert.py

```diff
@@ -110,32 +112,43 @@ def xml_nodes_to_dict(xml_nodes):
     return dict_nodes
 
 
-def save_dict_in_yaml_file(d: Dict, filename: str) -> None:
+def save_dict_in_yaml_file(d: Dict, stream) -> None:
     '''
-    Save a dict in filename as YAML
+    Write a dict as YAML in a stream
     '''
-    with open(filename, 'w', encoding='utf8') as f:
-        yaml.add_representer(str, yaml_str_presenter)
-        yaml.add_representer(dict, yaml_dict_presenter)
-        _yaml = yaml.dump(d, f, default_flow_style=False)
+    yaml.add_representer(str, yaml_str_presenter)
+    yaml.add_representer(dict, yaml_dict_presenter)
+    yaml.dump(d, stream, default_flow_style=False)
 
 
-def get_output_filename(args: argparse.PARSER, original_extension: str, target_extension: str) -> str:
+@contextmanager
+def get_output_stream(args: argparse.PARSER, original_extension: str, target_extension: str):
     '''
     Allow to retrieve the output filename from the arguments
     '''
+    stream = None
     if args.output:
-        return args.output
-    return args.filename.replace(original_extension, target_extension)
+        filename = '/dev/stdout' if '-' == args.output else args.output
+        stream = open(filename, 'w', encoding='utf8')
+    else:
+        stream = open(args.filename.replace(original_extension, target_extension), 'w', encoding='utf8')
+    yield stream
+    stream.close()
```

Comment:
What's the reason for the context manager here? The returned file object should already implement `__enter__()` and `__exit__()`, so I think the `with get_output_stream(...) as stream` below should work fine without this extra contextmanager decorator. For example:

```python
def f():
   return open('/tmp/abc', 'w')
with f() as stream:
   print(stream.closed)
print(stream.closed)
```

Returns:

```
False
True
```

Reply:
ahah ! Of course ! I didn't think about it but you are totally right ! Thanks :)